nixops/util.py

```diff
@@ -149,21 +150,33 @@ def _transform_value(key: Any, value: Any) -> Any:
             if not ann:
                 return value
 
-            if inspect.isclass(ann) and issubclass(ann, ImmutableValidatedObject):
+            if isclass(ann) and issubclass(ann, ImmutableValidatedObject):
                 value = ann(**value)
 
-            # Support Sequence[ImmutableValidatedObject]
-            if isinstance(value, tuple) and not isinstance(ann, str):
```

Comment:
This rejects strings, but in the new code all Sequences are matched, because

```python
>>> isinstance("hi", collections.abc.Sequence)
True
```

Could you exclude strings from the `match`, to be sure, or perhaps explain why strings don't occur here if that's the case?